exasol/ds/sandbox/lib/asset_printing/print_assets.py

```diff
@@ -237,41 +251,71 @@ def print_ec2_keys(aws_access: AwsAccess, filter_value: str, printing_factory: P
 
-    text_print = printing_factory.create_text_printer()
-
+    text_print = printing_factory.create_text_printer(console_only=True)
     text_print.print((TextObject("You can remove key-pairs using AWS CLI:\n"),
-                     TextObject("'aws ec2 delete-key-pair --key-pair-id "),
-                     HighlightedTextObject("KeyPairId"), TextObject("'")))
+                      TextObject("'aws ec2 delete-key-pair --key-pair-id "),
+                      HighlightedTextObject("KeyPairId"),
+                      TextObject("'")))
     text_print.print(tuple())
 
 
-def print_with_printer(aws_access: AwsAccess, asset_id: Optional[AssetId],
-                       asset_types: Tuple[str], filter_value: str, printing_factory: PrintingFactory):
-    if AssetTypes.AMI.value in asset_types:
-        print_amis(aws_access, filter_value, printing_factory)
-    if AssetTypes.SNAPSHOT.value in asset_types:
-        print_snapshots(aws_access, filter_value, printing_factory)
-    if AssetTypes.EXPORT_IMAGE_TASK.value in asset_types:
-        print_export_image_tasks(aws_access, filter_value, printing_factory)
-    if AssetTypes.VM_S3.value in asset_types:
-        print_s3_objects(aws_access, asset_id, printing_factory)
-    if AssetTypes.CLOUDFORMATION.value in asset_types:
-        print_cloudformation_stacks(aws_access, filter_value, printing_factory)
-    if AssetTypes.EC2_KEY_PAIR.value in asset_types:
-        print_ec2_keys(aws_access, filter_value, printing_factory)
-
-
-def print_assets(aws_access: AwsAccess, asset_id: Optional[AssetId], outfile: Optional[str],
-                 asset_types: Tuple[str] = all_asset_types()):
-    if asset_id is None:
-        filter_value = "*"
-    else:
-        filter_value = asset_id.tag_value
-
-    if outfile is not None:
-        with open(outfile, "w") as f:
-            printing_factory = MarkdownPrintingFactory(f)
-            print_with_printer(aws_access, asset_id, asset_types, filter_value, printing_factory)
-    else:
-        printing_factory = RichConsolePrintingFactory()
-        print_with_printer(aws_access, asset_id, asset_types, filter_value, printing_factory)
+def print_with_printer(
+        aws_access: AwsAccess,
+        asset_id: Optional[AssetId],
+        asset_types: Tuple[AssetTypes],
+        filter_value: str,
+        printing_factory: PrintingFactory,
+):
+    print_function = {
+        AssetTypes.DOCKER: print_docker_images,
+        AssetTypes.AMI: print_amis,
+        AssetTypes.SNAPSHOT: print_snapshots,
+        AssetTypes.EXPORT_IMAGE_TASK: print_export_image_tasks,
+        AssetTypes.VM_S3: print_s3_objects,
+        AssetTypes.CLOUDFORMATION: print_cloudformation_stacks,
+        AssetTypes.EC2_KEY_PAIR: print_ec2_keys,
+    }
+
+    def second_arg(asset_type: AssetTypes):
+        if asset_type in (AssetTypes.DOCKER, AssetTypes.VM_S3):
+            return asset_id
+        return filter_value
+
+    selected = (a for a in AssetTypes if a in asset_types)
```

Comment:
```suggestion
```

Reply:
The idea of this line is, to process the values passed in parameter `asset_types` but using the order defined by Enum `AssetTypes`.